src/agent/rustjail/src/mount.rs

```diff
@@ -1224,4 +1267,86 @@ mod tests {
 
         assert!(check_proc_mount(&mount).is_err());
     }
+    #[test]
+    fn test_rootfs_not_exists() {
+        let rootfs = "/home/rootfs";
+        assert_eq!("/home/rootfs/a/b/c", secure_join(rootfs, "a/b/c"));
+    }
+
+    #[test]
+    fn test_rootfs_relative() {
+        let rootfs = "/home/rootfs";
+        assert_eq!("/home/rootfs/a/b/c", secure_join(rootfs, "../../../a/b/c"));
```

Comment:
Could you add a few more test cases for `insecure_path` values of:

- `/`
- `//`, `//////`, `/../..////../..////../` (you get the idea ;)
- `/../`
- `/../../etc/`

Might be worth considering the following `root`-only tests too?

- `/proc/1/root/`
- `/proc/1/root/../`
- `/proc/1/root/../etc/`

I suggested the table-based approach as it makes adding new unit tests much easier - you don't need to keep inventing new test function names either ;)